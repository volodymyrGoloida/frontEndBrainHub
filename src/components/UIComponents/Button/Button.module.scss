@import "../../../UI-global-styles/mixins";
@import "../../../UI-global-styles/variables";

.buttonRoot {
  @include DisplayFlexMixin(column, center, center);
  width: 100%;
  height: 3rem;
  font-size: 1rem;
  box-sizing: border-box;
  background-color: $purple;
  border: none;
  cursor: pointer;
  transition-duration: 0.5s;
  border-radius: 0.25rem;
  color: white;
  &:focus {
    box-shadow: 0 0 0 2px #9eabd6;
  }
  &:hover {
    background-color: white;
    color: black;
  }
}
