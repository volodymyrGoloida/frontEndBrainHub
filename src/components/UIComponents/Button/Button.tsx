import React from "react";
import scss from "./Button.module.scss";

interface ButtonProps {
  children?: React.ReactNode;
}

const Button: React.FC<ButtonProps> = ({ children }) => {
  return <button className={scss.buttonRoot}>{children}</button>;
};
export default Button;
