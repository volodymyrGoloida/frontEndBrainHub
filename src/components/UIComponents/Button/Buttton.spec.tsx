import React from "react";
import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("component Button", () => {
  it("should successfully render Button", () => {
    const { baseElement } = render(<Button>some button</Button>);
    const button = screen.getByText("some button");
    expect(button).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
