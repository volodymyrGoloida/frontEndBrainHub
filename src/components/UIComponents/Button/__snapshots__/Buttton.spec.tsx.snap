// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Button should successfully render Button 1`] = `
<body>
  <div>
    <button
      class="buttonRoot"
    >
      some button
    </button>
  </div>
</body>
`;
